free_fleet_client_ros2/include/free_fleet/ros2/client_node.hpp

```diff
@@ -62,8 +62,9 @@ class ClientNode : public rclcpp::Node
 {
 public:
   using SharedPtr = std::shared_ptr<ClientNode>;
-  using ReadLock = std::unique_lock<std::mutex>;
-  using WriteLock = std::unique_lock<std::mutex>;
+  using Lock = std::shared_mutex;
```

Comment:
I would hesitate to call a mutex a lock due to semantic reasons. It might be a better idea to keep the named type `std::shared_mutex`.

Reply:
I agree, I just made this change to make it a lot easier to change and test. I could just call it `Mutex` which basically gives a better representation of what types of classes it could be.

Reply:
`Mutex` sounds like a good idea, thanks!